app/src/main/java/de/westnordost/streetcomplete/screens/about/LogsController.kt

```diff
@@ -0,0 +1,44 @@
+package de.westnordost.streetcomplete.screens.about
+
+import de.westnordost.streetcomplete.data.logs.LogLevel
+import de.westnordost.streetcomplete.data.logs.LogMessage
+import de.westnordost.streetcomplete.data.logs.LogsDao
+import de.westnordost.streetcomplete.util.ktx.toEpochMilli
+import de.westnordost.streetcomplete.util.logs.Log
+import kotlinx.datetime.LocalDateTime
+
+class LogsController(private val logsDao: LogsDao) {
+    fun getLogs(filters: LogsFilters): List<LogMessage> {
+        return logsDao.getAll(
+            levels = filters.levels,
+            messageContains = filters.messageContains,
+            newerThan = filters.timestampNewerThan?.toEpochMilli(),
+            olderThan = filters.timestampOlderThan?.toEpochMilli()
+        )
+    }
+
+    fun deleteOlderThan(timestamp: Long) {
+        val deletedCount = logsDao.deleteOlderThan(timestamp)
+        if (deletedCount > 0) {
+            Log.v(TAG, "Deleted $deletedCount old log messages")
+        }
+    }
+
+    companion object {
+        private const val TAG = "LogsController"
+    }
+}
+
+data class LogsFilters(
```

Comment:
This data class is curious. Why not pass the filter parameters directly? I cannot think of a reason why, other than that this data class is maybe used in the UI, a view model, if you will? In which case it might be slightly better to move this data class to the UI code.
It would be more consistent with the rest of the codebase.

Reply:
You're correct that this is kind of a view model that I've used in the UI code.
Moved it alongside the UI code where it's used.